consumer/nf_mangement.go

```diff
@@ -113,9 +125,19 @@ var SendRegisterNFInstance = func(nrfUri, nfInstanceId string, profile models.Nf
 	return prof, resouceNrfUri, retrieveNfInstanceId, err
 }
 
-func SendDeregisterNFInstance() (problemDetails *models.ProblemDetails, err error) {
+func SendDeregisterNFInstance(ctx context.Context) (*models.ProblemDetails, error) {
 	logger.ConsumerLog.Infof("[AMF] Send Deregister NFInstance")
 
+	ctx, span := tracer.Start(ctx, "HTTP DELETE nrf/nf-instances/{nfInstanceID}")
+	defer span.End()
+
+	span.SetAttributes(
+		attribute.String("http.method", "DELETE"),
+		attribute.String("nf.target", "nrf"),
+		attribute.String("net.peer.name", amf_context.AMF_Self().NrfUri),
+		attribute.String("amf.nf.id", amf_context.AMF_Self().NfId),
+	)
+
```

Comment:
I think there is no need to remove the return variable names from the function.
NOTE: I was unable to make the comment/suggestion to cover the appropriate place/range but it is applicable all the way to `var problemDetails *models.ProblemDetails`
```suggestion
func SendDeregisterNFInstance(ctx context.Context) (problemDetails *models.ProblemDetails, err error) {
	logger.ConsumerLog.Infof("[AMF] Send Deregister NFInstance")

	ctx, span := tracer.Start(ctx, "HTTP DELETE nrf/nf-instances/{nfInstanceID}")
	defer span.End()

	span.SetAttributes(
		attribute.String("http.method", "DELETE"),
		attribute.String("nf.target", "nrf"),
		attribute.String("net.peer.name", amf_context.AMF_Self().NrfUri),
		attribute.String("amf.nf.id", amf_context.AMF_Self().NfId),
	)

	amfSelf := amf_context.AMF_Self()
	// Set client and set url
	configuration := Nnrf_NFManagement.NewConfiguration()
	configuration.SetBasePath(amfSelf.NrfUri)
	client := Nnrf_NFManagement.NewAPIClient(configuration)

	var res *http.Response

	res, err = client.NFInstanceIDDocumentApi.DeregisterNFInstance(ctx, amfSelf.NfId)

```

Reply:
Done